R/conversion.R

```diff
@@ -0,0 +1,131 @@
+#'
+#' @param npc A single or vector of nets per capita
+#' @inheritParams access_to_npc
+#'
+#' @return Predicted access
+#' @export
+npc_to_access <- function(npc, type = "loess"){
+  if(!type %in% c("loess", "loess_extrapolate", "linear")){
+    stop("type must be one of: loess, loess_extrapolate or linear")
+  }
+  npc_fits <- get("npc_fits")
+  smooth <- npc_fits[[type]]
+  access <- seq(0, 1, 0.001)
+  pred <- access_to_npc(access, type)
+  access_out <- stats::approx(x = pred, y = access, xout = npc)$y
+  access_out
+}
+
+#' Convert usage to access
+#'
+#' @param usage A single value or vector of desired target usages to model.
+#' @param use_rate A single value or vector of usage rates.
+#'
+#' @return Access
+#' @export
+usage_to_access <- function(usage, use_rate){
+  if(any(usage < 0 | usage > 1, na.rm = TRUE)){
+    stop("usage must be between 0 and 1")
+  }
+  
+  access <- usage / use_rate
+  if(any(access > 1, na.rm = TRUE)){
+    warning("Target usage(s) cannot be achieved with input usage_rates - return NA")
+    access[access > 1] <- NA
+  }
+  return(access)
+}
+
+#' Convert access to usage
+#'
+#' @param access A single value or vector of access.
+#' @inheritParams usage_to_access
+#'
+#' @return Usage
+#' @export
+access_to_usage <- function(access, use_rate){
+  if(any(access < 0 | access > 1, na.rm = TRUE)){
+    stop("access must be between 0 and 1")
+  }
+  
+  usage <- access * use_rate
+  return(usage)
+}
+
+#' Convert nets per capita to annual nets per capital delivered
+#'
+#' @param npc Nets per capita
+#' @param distribution_freq A single distribution frequency of nets in days. Default = 3*365 (3 years).
+#' @param half_life Net loss half life (days)
+#' @param net_loss_function Option to choose between exponential net loss (net_loss_exp) or
+#' the smooth-compact net loss function from Bertozzi-Villa, Amelia, et al.
+#' Nature communications 12.1 (2021): 1-12 (net_loss_map). Default = net_loss_map.
+#' @param ... additional arguments for the net_loss_function
+#' 
+#' @return Annual nets per capita delivered
+#' @export
+npc_to_anpcd <- function(npc, distribution_freq, half_life, net_loss_function = net_loss_map, ...){
+  if(any(npc < 0, na.rm = TRUE)){
+    stop("npc must be > 0")
+  }
+  if(any(distribution_freq < 0, na.rm = TRUE)){
+    stop("distribution_freq must be > 0")
+  }
+  if(any(half_life < 0, na.rm = TRUE)){
+    stop("half_life must be > 0")
+  }
+  
+  npc * mapply(function(distribution_freq, half_life){
+    mean(net_loss_function(t = seq(0, distribution_freq, 1), half_life = half_life, ...))
+  }, distribution_freq = distribution_freq, half_life = half_life)
+}
+
+#' Convert annual nets per capital delivered to nets per capita
+#'
+#' @param anpcd Annual nets per capital delivered
+#' @inheritParams npc_to_anpcd
+#' 
+#' @return Nets per capita
+#' @export
+anpcd_to_npc <- function(anpcd, distribution_freq, half_life, net_loss_function = net_loss_map, ...){
```

Comment:
I think as package user I would find it better if everything was a bit more intuitive, rather than having to look up all the abbreviations (since all these net metric definitions are already quite confusing). I would at a minimum rename all columns in the final output to be fully understandable (nets per capita, annual nets distributed per capita) instead of using abbreviations